src/cpp/fastrtps_deprecated/transport/TCPTransportInterface.cpp

```diff
@@ -449,19 +449,27 @@ bool TCPTransportInterface::transform_remote_locator(
 {
     if (IsLocatorSupported(remote_locator))
     {
-        if (is_local_locator(remote_locator))
+        if (is_locator_allowed(remote_locator))
+        {
+            if (IPLocator::isLocal(remote_locator))
+            {
+                // Loopback locator
+                fill_local_ip(result_locator);
+                IPLocator::setPhysicalPort(result_locator, IPLocator::getPhysicalPort(remote_locator));
+                IPLocator::setLogicalPort(result_locator, IPLocator::getLogicalPort(remote_locator));
+                return true;
```

Comment:
I miss the checking that `localhost` is allowed too:

```suggestion
                if (is_locator_allowed(result_locator))
                {
                    return true;
                }
```

Reply:
This check should be added to the `else if` branch too.